### ISSUE TITLE
Holding shift while rotating text does not do the right thing

### ISSUE THREAD

**Issue by fsih** [LLK/scratch-paint#332]
This will be relevant once the textArea branch is merged.

Shift-rotate a text element:
![shifttext](https://user-images.githubusercontent.com/2855464/37779135-07219f86-2dc2-11e8-9d27-1b9ce41b086f.gif)

**Comment by Jacco** [comment]
Hi there,

I did some research on this one. I haven't found why this only goes wrong with TEXT and not with other SHAPES. But I found a fix for the issue in rotate-tool.js:

The fix also fixes #343 (since that issue is essentially a duplicate?)

Let me know if you want a PR for this one (it did not mention the 'HELP WANTED' label).

```
    onMouseDrag (event) {
        let rotAngle = (event.point.subtract(this.rotGroupPivot)).angle;
        
        for (let i = 0; i < this.rotItems.length; i++) {
            const item = this.rotItems[i];
            
            if (!item.data.origRot) {
                item.data.origRot = item.rotation;
            }
            
            if (event.modifiers.shift) {
                rotAngle = Math.round(rotAngle / 45) * 45;
            }
            item.rotate(rotAngle - this.prevRot[i], this.rotGroupPivot);
            this.prevRot[i] = rotAngle;
        }
    }
```

This code is a little simpler than the old code. Holding SHIFT now only changes the rotAngle and is handled identically with normal rotate.

Another point I found: The prevRot does not need to be an array, in how the routine currently works. If you want this simplified let me know. I can easily add it to the PR.

Regards, Jacco

**Comment by fsih** [comment]
@Jacco Nice! It's much simpler to see what's happening this way. I'm not totally sure why applyMatrix = false was originally added. It looks like it came from here: https://github.com/w00dn/papergrapher/commit/a038207cfc75442cc9ad0b311bb6565bc031b57a#diff-cc810f12347213d487c4055f3382028dR121 so it may have been needed for the rotation handles to work, and in Scratch we aren't using rotation handles currently, so it should be fine to remove.

That's a good point about prevRot too! Please do simplify it. Thanks for all your help!

**Comment by fsih** [comment]
Closed via #359
